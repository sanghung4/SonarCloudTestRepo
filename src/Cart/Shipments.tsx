import {
  Box,
  Collapse,
  FormControl,
  FormControlLabel,
  IconButton,
  RadioGroup,
  Typography,
  alpha,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { DeliveryMethodEnum, Maybe } from 'generated/graphql';
import { CloseIcon, WarningIcon } from 'icons';
import { radio } from 'utils/inputTestId';
import { useCartContext } from 'providers/CartProvider';

type ShipmentsProps = {
  isDisabled: boolean;
  deliveryMethod?: Maybe<DeliveryMethodEnum>;
  shouldShipFullOrder: string;
  setShouldShipFullOrder: (shouldShipFullOrder: string) => void;
  stockAlertOpen: boolean;
  setStockAlertOpen: (state: boolean) => void;
};

function Shipments(props: ShipmentsProps) {
  /**
   * Props
   */
  const isDelivery = props.deliveryMethod === DeliveryMethodEnum.Delivery;

  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { cart: { id: cartId } = { id: undefined }, updateDelivery } =
    useCartContext();

  return (
    <Collapse in={props.stockAlertOpen}>
      <Box
        bgcolor={(theme) => alpha(theme.palette.error.main, 0.05)}
        borderRadius="10px"
        p={3}
        mt={3}
      >
        <Box display="flex" alignItems="center" color="error.main">
          <Box
            component={WarningIcon}
            sx={{ verticalAlign: 'middle' }}
            mr={3}
          />
          <Box flex={1} display="flex" flexDirection="column">
            <Typography variant="h5" color="inherit">
              {t('cart.outOfStockAll')}
            </Typography>
            {isDelivery && (
              <>
                <Typography variant="caption" color="textPrimary">
                  {t('cart.outOfStockContinue')}
                </Typography>
                <Box pt={isSmallScreen ? 4 : 2} pb={1} color="text.primary">
                  <Box ml={1} mr={isSmallScreen ? -3 : 0}>
                    <FormControl component="fieldset">
                      <RadioGroup
                        row={!isSmallScreen}
                        aria-label="number-of-shipments"
                        name="shouldShipFullOrder"
                        value={props.shouldShipFullOrder}
                        onChange={handleShouldShipFullOrder}
                      >
                        <Box
                          component={FormControlLabel}
                          value="one"
                          control={radio('delivery-once-radio-button')}
                          label={t('cart.deliverOnce') as string}
                          disabled={props.isDisabled}
                          pr={isSmallScreen ? 0 : 10}
                          pb={isSmallScreen ? 2 : 0}
                        />
                        <Box
                          component={FormControlLabel}
                          value="multiple"
                          control={radio('delivery-multiple-radio-button')}
                          label={t('cart.deliverMultiple') as string}
                          disabled={props.isDisabled}
                        />
                      </RadioGroup>
                    </FormControl>
                  </Box>
                </Box>
              </>
            )}
          </Box>
          <Box
            component={IconButton}
            aria-label="close"
            color="inherit"
            size="small"
            data-testid="shipment-close-button"
            onClick={() => props.setStockAlertOpen(false)}
            m={-0.375}
          >
            <CloseIcon />
          </Box>
        </Box>
      </Box>
    </Collapse>
  );

  /**
   * Callback Definitions
   */
  function handleShouldShipFullOrder(
    event: React.ChangeEvent<HTMLInputElement>
  ) {
    if (cartId) {
      const { value } = event.target;
      props.setShouldShipFullOrder(value);
      updateDelivery?.({ cartId, shouldShipFullOrder: value === 'one' });
    }
  }
}

export default Shipments;
