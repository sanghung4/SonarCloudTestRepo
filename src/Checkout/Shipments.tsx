import { useState } from 'react';

import {
  FormControl,
  FormControlLabel,
  Grid,
  RadioGroup,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { shipmentStyles } from 'Checkout/util/styles';
import { radio } from 'utils/inputTestId';
import { useCartContext } from 'providers/CartProvider';

type Props = {
  shouldShipFullOrder: boolean;
};

export default function Shipments(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * State
   */
  const defaultShipment = props.shouldShipFullOrder ? 'one' : 'multiple';
  const [shipments, setShipments] = useState(defaultShipment);

  /**
   * Context
   */
  const { cart, updateDelivery } = useCartContext();

  /**
   * Render
   */
  return (
    <Grid container mb={8}>
      <Grid item xs={12} md={3}>
        <Typography
          variant="body1"
          color="textSecondary"
          pb={isSmallScreen ? 3 : 0}
        >
          {t('common.delivery')}
          <Typography variant="body1" component="span" color="error">
            {'*'}
          </Typography>
        </Typography>
      </Grid>
      <Grid item xs={12} md={9}>
        <FormControl component="fieldset" sx={shipmentStyles.formControlSx}>
          <RadioGroup
            aria-label="shipments"
            name="shipments"
            value={shipments}
            onChange={handleShipmentChange}
          >
            <FormControlLabel
              value="one"
              control={radio('delivery-once-radio-button')}
              label={`${t('cart.deliverOnceLong')}`}
              sx={shipmentStyles.formLabelOneSx}
            />
            <FormControlLabel
              value="multiple"
              control={radio('delivery-multiple-radio-button')}
              label={`${t('cart.deliverMultipleLong')}`}
              sx={shipmentStyles.formLabelMultipleSx}
            />
          </RadioGroup>
        </FormControl>
      </Grid>
    </Grid>
  );

  /**
   * Callback Definitions
   */
  function handleShipmentChange(event: React.ChangeEvent<HTMLInputElement>) {
    setShipments(event.target.value);
    if (cart)
      updateDelivery?.({
        cartId: cart.id,
        shouldShipFullOrder: event.target.value === 'one'
      });
  }
}
