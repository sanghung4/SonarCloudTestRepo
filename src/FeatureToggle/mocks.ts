import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
