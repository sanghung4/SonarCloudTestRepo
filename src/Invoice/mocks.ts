import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
