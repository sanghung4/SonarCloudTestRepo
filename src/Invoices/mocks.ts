import { subMonths } from 'date-fns';
import { Invoice, InvoicesDocument } from 'generated/graphql';
import { mockRow } from 'test-utils/mockTableInstance';
import { formatDate } from 'utils/dates';

export const success = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'ECLIPSE'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Current',
            customerPo: '1096872',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const successMincron = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'MINCRON'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '6566',
            jobName: 'NEP 7 ASHVELLE'
          },
          {
            age: 'Current',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '56787',
            jobName: 'WALTER HORN'
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'SHOP',
            jobName: 'SEQUAYESH TR'
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'RETREAT',
            jobName: 'RETREAT AT ARDEN'
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'AVLNEP7',
            jobName: 'WALTER HORN'
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'AIKENON',
            jobName: 'AIKEN AVL'
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const mockTestTableMemo = {
  invoices: {
    bucketNinety: 0.0,
    bucketOneTwenty: 0.0,
    bucketSixty: 0.0,
    bucketThirty: 0.0,
    bucketFuture: 0.0,
    currentAmt: 35224.96,
    invoices: [
      {
        age: 'Future',
        customerPo: '109687',
        invoiceDate: '05/02/2022',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227972.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '6566',
        jobName: 'NEP 7 ASHVELLE'
      },
      {
        age: 'Current',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227975.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '56787',
        jobName: 'WALTER HORN'
      },
      {
        age: 'Over120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220294.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'SHOP',
        jobName: 'SEQUAYESH TR'
      },
      {
        age: '91-120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110222345.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'RETREAT',
        jobName: 'RETREAT AT ARDEN'
      },
      {
        age: 'Deposit',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220974.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AVLNEP7',
        jobName: 'WALTER HORN'
      },
      {
        age: '31-60',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227973.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AIKENON',
        jobName: 'AIKEN AVL'
      }
    ],
    totalAmt: 35232.27,
    totalPastDue: 7.3085938
  }
};

const rowToMock = success.result.data.invoices.invoices[0];
export const mockInvoiceRow = mockRow<Invoice>(rowToMock, rowToMock, {});
