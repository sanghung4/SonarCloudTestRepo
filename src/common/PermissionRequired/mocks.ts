import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
