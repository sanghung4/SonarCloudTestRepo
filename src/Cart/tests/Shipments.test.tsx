import { act, fireEvent } from '@testing-library/react';

import Shipments from 'Cart/Shipments';
import { mockCartContext } from 'Cart/tests/mocks';
import { CartContext } from 'providers/CartProvider';
import { render } from 'test-utils/TestWrapper';

const mockCartContextVal = { ...mockCartContext };

describe('Cart - Shipments', () => {
  it('Should match snapshot when it is closed', async () => {
    const wrapper = render(
      <CartContext.Provider value={mockCartContextVal}>
        <Shipments
          isDisabled={false}
          stockAlertOpen={false}
          shouldShipFullOrder=""
          setShouldShipFullOrder={jest.fn()}
          setStockAlertOpen={jest.fn()}
        />
      </CartContext.Provider>
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match snapshot with normal data', async () => {
    const wrapper = render(
      <CartContext.Provider value={mockCartContextVal}>
        <Shipments
          isDisabled={false}
          stockAlertOpen={true}
          shouldShipFullOrder="test"
          setShouldShipFullOrder={jest.fn()}
          setStockAlertOpen={jest.fn()}
        />
      </CartContext.Provider>
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect X to be called when clicking on the close button', async () => {
    const closeFn = jest.fn();
    const wrapper = render(
      <CartContext.Provider value={mockCartContextVal}>
        <Shipments
          isDisabled={false}
          stockAlertOpen={true}
          shouldShipFullOrder="test"
          setShouldShipFullOrder={jest.fn()}
          setStockAlertOpen={closeFn}
        />
      </CartContext.Provider>
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    fireEvent.click(wrapper.getByTestId('shipment-close-button'));
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(closeFn).toBeCalledWith(false);
  });
});
