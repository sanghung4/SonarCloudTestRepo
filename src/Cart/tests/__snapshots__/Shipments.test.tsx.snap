// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart - Shipments Should match snapshot when it is closed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-1q7njgz"
            >
              <div
                class="MuiBox-root css-13rkfht"
              >
                <svg
                  class="MuiBox-root css-1l04f3i"
                  sx="[object Object]"
                >
                  warning.svg
                </svg>
                <div
                  class="MuiBox-root css-k0sogd"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-oplc8n-MuiTypography-root"
                  >
                    One or more items are out of stock or below desired quantity
                  </h5>
                </div>
                <button
                  aria-label="close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiBox-root css-1p2d2y2-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="shipment-close-button"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiBox-root css-1q7njgz"
          >
            <div
              class="MuiBox-root css-13rkfht"
            >
              <svg
                class="MuiBox-root css-1l04f3i"
                sx="[object Object]"
              >
                warning.svg
              </svg>
              <div
                class="MuiBox-root css-k0sogd"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-oplc8n-MuiTypography-root"
                >
                  One or more items are out of stock or below desired quantity
                </h5>
              </div>
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiBox-root css-1p2d2y2-MuiButtonBase-root-MuiIconButton-root"
                data-testid="shipment-close-button"
                tabindex="0"
                type="button"
              >
                <svg>
                  delete.svg
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cart - Shipments Should match snapshot with normal data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-1q7njgz"
            >
              <div
                class="MuiBox-root css-13rkfht"
              >
                <svg
                  class="MuiBox-root css-1l04f3i"
                  sx="[object Object]"
                >
                  warning.svg
                </svg>
                <div
                  class="MuiBox-root css-k0sogd"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-oplc8n-MuiTypography-root"
                  >
                    One or more items are out of stock or below desired quantity
                  </h5>
                </div>
                <button
                  aria-label="close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiBox-root css-1p2d2y2-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="shipment-close-button"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiBox-root css-1q7njgz"
          >
            <div
              class="MuiBox-root css-13rkfht"
            >
              <svg
                class="MuiBox-root css-1l04f3i"
                sx="[object Object]"
              >
                warning.svg
              </svg>
              <div
                class="MuiBox-root css-k0sogd"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-oplc8n-MuiTypography-root"
                >
                  One or more items are out of stock or below desired quantity
                </h5>
              </div>
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiBox-root css-1p2d2y2-MuiButtonBase-root-MuiIconButton-root"
                data-testid="shipment-close-button"
                tabindex="0"
                type="button"
              >
                <svg>
                  delete.svg
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
